### PR TITLE
Modifies logic to hide datepicker when clicking browser back button

### PR DESCRIPTION
**Pull request by agondray**
- moved logic into the show() function in picker.js
 - used setTimeout and clearInterval instead of the onhashchange event
  * reason: onhashchange event only listens to urls containing a `#` symbol.

### CODE REVIEW

File: picker.js [hunk 1]
```diff
@@ -129,6 +132,16 @@ class Picker {
       document.addEventListener(`mousedown`, this.removeClickOut);
       document.addEventListener(`touchstart`, this.removeClickOut);
     }, 500);
+
+    // when used in a single-page app  or otherwise,
+    // hide datepicker when the browser's back button is pressed
+    let currentPage = window.location.href;
+    compareHrefInterval = setInterval(() => {
+      console.log('in setinterval');
```
Comment (kylealwyn) [inline picker.js]: no logs 🙃

File: picker.js [hunk 2]
```diff
@@ -129,6 +132,16 @@ class Picker {
...
+      if (currentPage != window.location.href) {
+        thePicker.hide();
```
Comment (kylealwyn) [inline picker.js]: use `this.hide()` to reference the instance of the picker `show` is being called on

File: picker.js [hunk 3]
```diff
@@ -129,6 +132,16 @@ class Picker {
       document.addEventListener(`mousedown`, this.removeClickOut);
       document.addEventListener(`touchstart`, this.removeClickOut);
     }, 500);
+
+    // when used in a single-page app  or otherwise,
+    // hide datepicker when the browser's back button is pressed
+    let currentPage = window.location.href;
+    compareHrefInterval = setInterval(() => {
+      console.log('in setinterval');
+      if (currentPage != window.location.href) {
```
Comment (kylealwyn) [inline picker.js]: use triple equality (`!==`, `===`) where you can

File: picker.js [hunk 4]
```diff
@@ -129,6 +132,16 @@ class Picker {
       document.addEventListener(`mousedown`, this.removeClickOut);
       document.addEventListener(`touchstart`, this.removeClickOut);
     }, 500);
+
+    // when used in a single-page app  or otherwise,
+    // hide datepicker when the browser's back button is pressed
+    let currentPage = window.location.href;
+    compareHrefInterval = setInterval(() => {
```
Comment (kylealwyn) [inline picker.js]: did you try the `onpopstate` listener? may be cleaner then this polling setup:  https://developer.mozilla.org/en-US/docs/Web/API/WindowEventHandlers/onpopstate

**Review by kylealwyn — CHANGES_REQUESTED** [review]

File: picker.js (hunk 1 above)
Comment (agondray) [reply picker.js]: thanks for catching that! 👍

File: picker.js (hunk 2 above)
Comment (agondray) [reply picker.js]: 👍

File: picker.js (hunk 4 above)
Comment (agondray) [reply picker.js]: 👍 onpopstate works! thanks for the suggestion.

File: picker.js (hunk 3 above)
Comment (agondray) [reply picker.js]: gotcha. i'll do my best to remember to use triple equality moving forward.

**Review by tirodactyl — APPROVED** [review]

**Review by kylealwyn — APPROVED** [review]

**Pull request merged**